src/libs/OptionsListUtils.ts

```diff
@@ -23,41 +34,138 @@ import * as TaskUtils from './TaskUtils';
 import * as TransactionUtils from './TransactionUtils';
 import * as UserUtils from './UserUtils';
 
+type Tag = {
+    enabled: boolean;
+    name: string;
+    accountID: number | null;
+};
+
+type Option = Partial<ReportUtils.OptionData>;
+
+type PayeePersonalDetails = {
+    text: string;
+    alternateText: string;
+    icons: OnyxCommon.Icon[];
+    descriptiveText: string;
+    login: string;
+    accountID: number;
+};
+
+type CategorySection = {
+    title: string | undefined;
+    shouldShow: boolean;
+    indexOffset: number;
+    data: Option[];
+};
+
+type Category = {
+    name: string;
+    enabled: boolean;
+};
+
+type Hierarchy = Record<string, Category & {[key: string]: Hierarchy & Category}>;
```

Comment:
How about extracting:
```suggestion
type X = Record<string, Hierarchy & Category>;
type Hierarchy = Record<string, Category & X>;
```

Reply:
This is not working the `DeepRecord` produces type like that `type Hierarchy = {
    [key: string]: Category | DeepRecord<string, Category>;
}` but I need something like that 
```
type Hierarchy = {
name: string;
enabled: boolean;
[key]: {
    name: string;
    ....
  }
}